### ISSUE TITLE
Extra link at dropdown item

### ISSUE THREAD

**Issue by agharium** [yairEO/tagify#530]
Hi, I need to set another link/button on every dropdown item like this:

![image](https://user-images.githubusercontent.com/28984633/83408820-a8a03300-a3e9-11ea-860e-28f13ab880ef.png)

I've tried doing it by "hacking" the DOM, but it didn't worked at all:

![image](https://user-images.githubusercontent.com/28984633/83408993-ef8e2880-a3e9-11ea-9e97-43d14476ca15.png)

If I click on it, it just detects it as a click on the dropdown-item value itself, adding the value to the tagify input.

This is the code I used for the "hack" attempt (it's an ugly code):

```
tagify.on('dropdown:show',function(e){
    $('.tagify__dropdown__item').each(function(){
        $(this).html($(this).html() + '</div><a href="https://google.com.br" target="blank" class="float-right" style="z-index: 100000"> Remover </a><div>')
        console.log($(this).parent().html())
    })
    
    $('.tagify__dropdown__item').unbind('click');

    $('.tagify__dropdown__item .btn').each(function(){
        $(this).click(function() {
            alert("test")
        })
    })
})
```

Could I get some help on doing this?..

**Comment by yairEO** [comment]
look at the TEMPLATES section in Tagify's README (in the repo), it explains
just that

**Comment by Attrash-Islam** [comment]
Stopping the propagation may fix your issue on the option selection.

**Comment by yairEO** [comment]
Still need to think of a way to actually attach an onClick callback to anything inside the template in a smart & easy way.

**Comment by agharium** [comment]
> Still need to think of a way to actually attach an `onClick` callback to anything inside the template in a smart & easy way.

Yeah, I did it in the template way (ty for pointing that out), but still got the "inner callback"  on my way. It never gets activated, but the outer dropdown item is always activated no matter what.

**Comment by yairEO** [comment]
I've update the [demo](https://jsbin.com/henoxux/edit?css,js,output) and it's working, but there are a few things I need to change internally in Tagify to make the UX better, for example, not close the dropdown when clicking on an *action button* inside a suggestion item, and allow easier event binding for those buttons' actions.

**Comment by agharium** [comment]
> I've update the [demo](https://jsbin.com/henoxux/edit?css,js,output) and it's working, but there are a few things I need to change internally in Tagify to make the UX better, for example, not close the dropdown when clicking on an _action button_ inside a suggestion item, and allow easier event binding for those buttons' actions.

Hi, I forgot to reply to this.. indeed it worked fine, but I failed to "plug it and play" on my code. It didn't immediately worked for me. For lack of time, I couldn't dedicate more time onto it, probably something on my code was messing with your solution. 

Now, I see that you committed a fix related to this issue, but I'm still struggling. The previous solution u sent me ain't working anymore, and I also can't figure out how to use the (new, I believe) `onSugggestionClick` hook. I'm messing with the hook README's demo (`beforeRemoveTag`), but no luck (I'm probably not too proficient with JS :neutral_face:)

Could you shed me a light over this?..

**Comment by agharium** [comment]
Well, it's working perfectly, except for still adding the value when clicking the remove button.

So I did a massive and ugly workaround: :blush:

```
function removeWhitelistItem(value) {
    var index = tagify.settings.whitelist.indexOf(value);
    if( value && index > -1 )
        tagify.settings.whitelist.splice(index, 1)
    
    tagify.removeTags(value)
            
    removerNomeJson(nome)
}
```

**Comment by yairEO** [comment]
Now it works good (pushed new Tagify version `3.11.1`)

https://jsbin.com/tuwihuf/edit?html,js,output

**Comment by agharium** [comment]
Thank you, brother. Updated my code with the new version - no more ugly workaround, nor hiccups! :sweat_smile:

Wonderful plugin, just great. Congrats on your work.

 Wish you success and God's blessings.
